operator/main.go

```diff
@@ -148,6 +161,74 @@ func runOperator(cmd *cobra.Command) {
 	}
 	close(k8sInitDone)
 
+	k8sversion.Update(k8s.Client(), option.Config)
+	if !k8sversion.Capabilities().MinimalVersionMet {
+		log.Fatalf("Minimal kubernetes version not met: %s < %s",
+			k8sversion.Version(), k8sversion.MinimalVersionConstraint)
+	}
+
+	// Get hostname for identity name of the lease lock holder.
+	// We identify the leader of the operator cluster using hostname.
+	operatorID, err := os.Hostname()
+	if err != nil {
+		log.WithError(err).Fatal("failed to get hostname")
```

Comment:
Nit: this msg might be cryptic when looking through logs when debugging; it's a bit too generic. Maybe something like:

 ```suggestion
		log.WithError(err).Fatal("Failed to get hostname when generating lease lock identity")
```